projects/ngx-sub-form/src/lib/ngx-sub-form.component.ts

```diff
@@ -66,6 +74,63 @@ export abstract class NgxSubFormComponent<ControlInterface, FormInterface = Cont
 
   private controlKeys: (keyof FormInterface)[] = [];
 
+  // instead of having the validators defined in the utils file
+  // we define them here to have access to the form types
+
+  // `ngxSubFormValidators` should be used at a group level validation
+  // with the `getFormGroupControlOptions` hook
+  protected ngxSubFormValidators = {
+    oneOf(keysArray: (keyof FormInterface)[][]): TypedValidatorFn<FormInterface> {
```

Comment:
why is this an array of arrays?

Reply:
Because one form component may have multiple properties that are polymorphic.

Example:
```
interface User {
  id: string;
  vehicle: MotorBike | Car | Truck;
  animal: Cat | Dog;
}
```

In the case we want to handle them without breaking down into sub components we can do:

```
oneOf([
  ['motorBikeVehicle', 'CarVehicle', 'TruckVehicle'],
  ['CatVehicle', 'DogVehicle']
])
```